kahuna/public/js/upload/jobs/upload-jobs.js

```diff
@@ -60,6 +66,21 @@ jobs.controller('UploadJobsCtrl',
     // this needs to be a function due to the stateful `jobItem`
     ctrl.jobImages = () => ctrl.jobs.map(jobItem => jobItem.image);
 
+    ctrl.onDeleteSuccess = function (resp, image) {
+        var index = ctrl.jobs.findIndex(i => i.image.data.id === image.data.id);
+
+        if (index > -1) {
+            ctrl.jobs.splice(index, 1);
+        }
+    };
+
+    ctrl.onDeleteError = function (err) {
+        if (err.body.errorKey === 'image-not-found') {
+            $state.go('upload', {}, {reload: true});
```

Comment:
But the deletion would have failed, so when the page is refreshed/you view your uploads, the image you thought you deleted somehow reappears. Might be confusing?